feat: various sdk improvements (#1313)
yetanotherco/aligned_layer: docs/3_guides/1.2_SDK_api_reference.md

- `[u8; 32]` - A 32-byte array representing the keccak256 hash of the verification key.

### `get_nonce_from_ethereum`

> Returns the nonce to use for a given address.

Reviewer: ```suggestion
Returns the nonce for a given address in Ethereum from aligned payment service contract.
Note that it might be out of sync if you recently sent proofs. For that see [`get_nonce_from_batcher`](#get_nonce_from_batcher)
```

- `EthereumCallError` if there is an error in the Ethereum call.
- `NotAContract(address)` if you are trying to send to an address that is not a contract. This generally occurs if you have misconfigured the `environment` parameter.

### `get_nonce_from_batcher`

Returns the next nonce for a given address from the batcher.

You should prefer this method instead of [`get_nonce_from_ethereum`](#get_nonce_from_ethereum) if you have recently sent proofs,

> as the batcher proofs might not yet be on ethereum, producing an out-of-sync nonce with the payment service contract on ethereum
> 

Reviewer: ```suggestion
as the batcher proofs might not yet be on ethereum, producing an out-of-sync nonce with the payment service contract on ethereum

```rust
pub async fn get_nonce_from_batcher(
    batcher_ws_url: &str,
    address: Address,
) -> Result<U256, GetNonceError> {
```
```

> # Arguments

Reviewer: ```suggestion
#### Arguments
```

- `batcher_url` - The batcher websocket url.
- `address` - The user address for which the nonce will be retrieved.

> # Returns

Reviewer: ```suggestion
#### Returns
```

- `Result<U256, NonceError>` - The next nonce of the proof submitter account.

> # Errors

Reviewer: ```suggestion
#### Errors
```

- `EthereumProviderError` if there is an error in the connection with the RPC provider.
- `EthereumCallError` if there is an error in the Ethereum call.

### `deposit_to_aligned`

Funds the batcher payment service in name of the signer

pub async fn deposit_to_aligned(
amount: U256,
signer: SignerMiddleware<Provider<Http>, LocalWallet>,
network: Network,

> ) -> Result<ethers::types::TransactionReceipt, errors::PaymentError> {

Reviewer: This code should be wrapped in a Rust code block like this:
```Rust
pub async fn deposit_to_aligned(
amount: U256,
signer: SignerMiddleware<Provider<Http>, LocalWallet>,
network: Network,
) -> Result<ethers::types::TransactionReceipt, errors::PaymentError> {
